plugin/lspconfig.lua

```diff
@@ -0,0 +1,85 @@
+local version_info = vim.version()
+if vim.fn.has 'nvim-0.7' ~= 1 then
+  local warning_str = string.format(
+    '[lspconfig] requires neovim 0.7 or later. Detected neovim version: 0.%s.%s',
+    version_info.minor,
+    version_info.patch
+  )
+  vim.notify_once(warning_str)
+  return
+end
+
+local util = require 'lspconfig.util'
+local configs = require 'lspconfig.configs'
+
+_G.lsp_complete_configured_servers = function()
+  return table.concat(vim.tbl_keys(configs), '\n')
+end
+
+_G.lsp_get_active_client_ids = function()
+  return table.concat(
+    vim.tbl_map(function(client)
+      return ('%d (%s)'):format(client.id, client.name)
+    end, require('lspconfig.util').get_managed_clients()),
+    '\n'
+  )
+end
+
+local get_clients_from_cmd_args = function(arg)
+  local result = {}
+  for id in (arg or ''):gmatch '(%d+)' do
+    result[id] = vim.lsp.get_client_by_id(tonumber(id))
+  end
+  if vim.tbl_isempty(result) then
+    return util.get_managed_clients()
+  end
+  return vim.tbl_values(result)
+end
+
+-- Called from plugin/lspconfig.vim because it requires knowing that the last
+-- script in scriptnames to be executed is lspconfig.
+vim.api.nvim_create_user_command('LspInfo', function()
+  require 'lspconfig.ui.lspinfo'()
+end, {
+  desc = 'Displays attached, active, and configured language servers',
+})
+
+vim.api.nvim_create_user_command('LspStart', function(info)
+  local server_name = info.args[1] ~= '' and info.args
+  if server_name then
+    if configs[server_name] then
+      configs[server_name].launch()
+    end
+  else
+    local other_matching_configs = require('lspconfig.util').get_other_matching_providers(vim.bo.filetype)
+    for _, config in ipairs(other_matching_configs) do
+      config.launch()
+    end
+  end
+end, {
+  desc = 'Manually launches a language server',
+  nargs = '?',
+  complete = 'custom,v:lua.lsp_complete_configured_servers',
```

Comment:
This can be a lua function, no need to create global variables. I don't remember if this was supported right from the start, or was implement in later neovim versions though.
```suggestion
  complete = lsp_complete_configured_servers,
```

Reply:
This was the initial state of the PR. @williamboman requested the use of `custom, v:lua` method

Reply:
Okay, I think the issue was that you didn't filter completions, like this:
```lua
function complete(arg)
  local res = {}
  for _, entry in ipairs(lsp_complete_configured_servers()) do
    if entry:sub(1, #arg) == arg then
      table.insert(res, entry)
    end
  end
  return res
end
```
This can happen inside `lsp_complete_configured_servers` of course.

Reply:
The original functions were these
```lua
local lsp_get_active_client_ids = function()
  return vim.tbl_map(function(client)
    return ('%d (%s)'):format(client.id, client.name)
  end, require('lspconfig.util').get_managed_clients())
end
```
and 

```lua
  desc = 'Manually launches a language server',
  nargs = '?',
  complete = function()
    return vim.tbl_keys(configs)
  end,
```

Reply:
Yes, you need to filter them then.

Reply:
Ah, I wasn't aware you'd have to implement the filtering yourself, that explains it. Maybe this can be rectified in a later PR, this is starting to go a bit in circles with people chiming in haha

Reply:
Yes, getting this merged and then adding improvements if necessary maybe a good idea!